joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -116,6 +116,29 @@ controller_interface::return_type JointTrajectoryController::update(
     return controller_interface::return_type::OK;
   }
 
+  // current state update
+  state_current_.time_from_start.set__sec(0);
+  read_state_from_hardware(state_current_);
+
+  if (start_holding_)
```

Comment:
Shouldn't we here check which interfaces are used?

Moreover, what I am missing here is if starting holding will work when trajectory is aborted.

Reply:
Just discussing this with @bmagyar, can we move this part into `set_hold_position`. The idea is that `set_hold_position` sets an adequate trajectory to enable this functionality. Ideally, we shouldn't have to change anything in the `update()` (or just a bit).

I would really like to avoid this `ıf`, i.e., avoid having another logic path – of course, if this is possible.

Reply:
I'll have a look

Reply:
I refactored it -> looks way cleaner now